packages/trader/src/AppV2/Components/TradeParameters/Duration/container.tsx

```diff
@@ -0,0 +1,126 @@
+            const timeDifference = +date - +current_date;
+            const dayDifference = Math.ceil(timeDifference / (1000 * 60 * 60 * 24));
+            setSelectedTime([dayDifference]);
+        };
+
+        useEffect(() => {
+            if (duration_unit === 'm' && duration > 59) {
+                const hour = Math.floor(duration / 60);
+                const minutes = duration % 60;
+                setSelectedHour([hour, minutes]);
+                setUnit('h');
+            }
+        }, [duration, duration_unit, setSelectedHour]);
+
+        const setWheelPickerValue = (index: number, value: string | number) => {
+            if (unit === 'h') {
+                const arr = selected_hour;
+                arr[index] = Number(value);
+                setSelectedHour(arr);
+            } else if (unit == 'd') {
+                setSelectedTime([Number(value)]);
+                const updated_date = new Date();
+                updated_date.setDate(updated_date.getDate() + Number(value));
+                setExpiryDate(updated_date);
+            } else {
+                setSelectedTime([Number(value)]);
+            }
+        };
+
+        return (
+            <div className='duration-container'>
+                <ActionSheet.Header title={<Localize i18n_default_text='Duration' />} />
+                <DurationChips duration_units_list={duration_units_list} onChangeUnit={onChangeUnit} unit={unit} />
+                <DurationWheelPicker
+                    unit={unit}
+                    setEndTime={val => {
+                        setEndTime(val);
+                    }}
+                    setWheelPickerValue={setWheelPickerValue}
+                    selected_hour={selected_hour}
+                    selected_time={selected_time}
+                    toggle_picker={toggle_picker}
+                />
+                {unit == 'd' && (
+                    <DurationEndTimePicker
+                        setExpiryDate={handleSelectExpiryDate}
+                        setSelectedTime={val => {
+                            setSelectedTime(val);
+                            setTogglePicker(!toggle_picker);
+                        }}
+                        expiry_date={expiry_date_data}
+                    />
+                )}
+                {unit == 'et' && (
+                    <div className='duration-container__endtime'>
+                        <CaptionText color='quill-typography__color--subtle'>
+                            <Localize i18n_default_text='Current time' />
+                        </CaptionText>
+                        <Text>{`${end_time} GMT`}</Text>
+                    </div>
+                )}
+                <ActionSheet.Footer
+                    alignment='vertical'
+                    primaryAction={{
+                        content: <Localize i18n_default_text='Save' />,
+                        onAction: () => {
+                            if (unit == 'h') {
```

Comment:
Not a big deal, but maybe it'll be better to create a named function for this onAction prop?